Mathlib/Algebra/BigOperators/Fin.lean

```diff
@@ -484,6 +484,32 @@ theorem prod_ofFn {n : ℕ} {f : Fin n → α} : (ofFn f).prod = ∏ i, f i := b
 #align list.prod_of_fn List.prod_ofFn
 #align list.sum_of_fn List.sum_ofFn
 
+@[to_additive]
+theorem prod_finProd {n : ℕ} {m : ℕ} {f : ℕ × ℕ → α} :
+    ∏ x : Fin n × Fin m, f (x.1, x.2) = ∏ x : Fin n, ∏ y : Fin m, f (x, y) :=
+  prod_finset_product univ univ (fun _ ↦ univ) <|
+      fun p => ⟨fun _ => ⟨mem_univ p.1, mem_univ p.2⟩, fun _ => mem_univ p⟩
```

Comment:
This should be
```suggestion
@[to_additive Fintype.sum_prod]
theorem Fintype.prod_prod {α β} [Fintype α] [Fintype β] [CommMonoid γ] {f : α × β → γ} :
    ∏ x : α × β, f x = ∏ x : α, ∏ y : β, f (x, y) :=
  prod_finset_product univ univ (fun _ ↦ univ) <|
      fun p => ⟨fun _ => ⟨mem_univ p.1, mem_univ p.2⟩, fun _ => mem_univ p⟩
```
and belongs in a different file. I'm pretty sure we already have it soemwhere...

Reply:
ok, great

Reply:
Are you planning to open a new PR that adds this in the right place? Or did you find the lemma after all?

Reply:
Oh, I just trusted you that we have it somewhere and was planning to look later where exactly. If I don't find it, I will re-open this pull request, moving and adapting the files...

Reply:
I didn't find it (but that could also be lack of searching skills).

So I added it in the place you suggested. 
One thing still to fix is the fact that `Fintype (α₁ × α₂)` is not interred automatically...

Reply:
Maybe @YaelDillies  can tell us where this lemma is hiding.

Reply:
https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there-code-for-X.3F/topic/group.20sum.2Fprod.20in.20pairs.20even.20and.20odd.20summands

Reply:
I mean this is defeq to `Finset.prod_product`.

Reply:
Does this mean we want to have those nonetheless?

Reply:
There is a slight inconsistency: `Finset.prod_product` has implicit arguments only, while `Finset.prod_disj_sum` has three explicit arguments. Don't know if this is intentional.